src/portable/st/synopsys/synopsys_common.h

```diff
@@ -0,0 +1,1448 @@
+#include "stdint.h"
```

Comment:
ah right, this is quite a pain that GD32V name register differently despite they are pretty much the same thing. I am ok adding this for gd32, some other non-stm may need this as well since synopsys dwc controller is quite popular. There is only one issue though, I think most of the definition is from ST driver, we should keep their license in this file, luckily it is compatible with MIT so we are all good.

Reply:
Perfect, I wasn't sure if you would be fine with this "hack". I have added the stm32 license header.